compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/transformers/body/resolve/FirExpressionsResolveTransformer.kt

```diff
@@ -691,19 +693,40 @@ open class FirExpressionsResolveTransformer(transformer: FirBodyResolveTransform
             (annotationCall.transformChildren(transformer, data) as FirAnnotationCall).also {
                 // TODO: it's temporary incorrect solution until we design resolve and completion for annotation calls
                 it.argumentList.transformArguments(integerLiteralTypeApproximator, null)
-                if (it.arguments.any { arg -> arg is FirNamedArgumentExpression }) {
-                    annotationCall.getCorrespondingConstructorReferenceOrNull(session)?.let { calleeReference ->
-                        val argumentMapping =
-                            mapArguments(it.arguments, calleeReference.resolvedSymbol.fir as FirFunction<*>)
-                                .toArgumentToParameterMapping()
+                annotationCall.getCorrespondingConstructorReferenceOrNull(session)?.let { calleeReference ->
+                    val callee = calleeReference.resolvedSymbol.fir as FirFunction<*>
+                    val argumentMapping = mapArguments(it.arguments, callee).toArgumentToParameterMapping()
+                    val varargParameter = callee.valueParameters.firstOrNull { param -> param.isVararg }
+                    if (varargParameter == null) {
                         it.replaceArgumentList(buildResolvedArgumentList(argumentMapping))
+                    } else {
+                        // TODO: refactor/reuse [Candidate#handleVarargs] in [FirCallCompletionResultsWriterTransformer]
```

Comment:
I will do it to avoid code duplication

Reply:
Found https://github.com/JetBrains/kotlin/commit/07b0ffef8473d7c8518a3a24f753605bdfd9cc8a. Thanks!